Login and Logout views
----------------------

We have an app instance:

    >>> from zope.testbrowser.testing import Browser
    >>> manager = Browser()
    >>> manager.handleErrors = False
    >>> manager.addHeader('Authorization', 'Basic manager:schooltool')
    >>> manager.open('http://localhost/')
    >>> 'SchoolTool' in manager.contents
    True

We add a person

    >>> manager.getLink('Persons').click()
    >>> manager.getLink('New Person').click()

    >>> manager.getControl('Full name').value = 'Some User'
    >>> manager.getControl('Username').value = 'user'
    >>> manager.getControl('Password').value = 'secret'
    >>> manager.getControl('Verify password').value = 'secret'
    >>> manager.getControl('Add').click()

We have the views ``@@login.html`` and ``@@logout.html`` on the main
application object. Those views log in or log out the user of the session.

    >>> anonymous = Browser()
    >>> anonymous.open('http://localhost/')
    >>> anonymous.getLink('Log In').click()
    >>> anonymous.headers['status']
    '200 Ok'

Make sure the current version number is somewhere on the page:

    >>> import schooltool
    >>> schooltool.VERSION in anonymous.contents
    True

Incorrect password:

    >>> anonymous.getControl('Username').value = 'user'
    >>> anonymous.getControl('Password').value = 'bad'
    >>> anonymous.getControl('Log in').click()
    >>> 'Username or password is incorrect' in anonymous.contents
    True

Another attempt:

    >>> anonymous.getControl('Username').value = 'user'
    >>> anonymous.getControl('Password').value = 'secret'
    >>> anonymous.getControl('Log in').click()

Now we are authenticated:

    >>> 'Some User' in anonymous.contents
    True

Now we can log out:

    >>> anonymous.getLink('Log Out').click()

We are logged out:

    >>> anonymous.getLink('Log in')
    Traceback (most recent call last):
    ...
    LinkNotFoundError
